hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/ingest/IngestWorkflowImpl.java

```diff
@@ -15,20 +15,149 @@
+    private final Supplier<AutoCloseableWrapper<HederaState>> stateAccessor;
+    private final WorkflowOnset onset;
+    private final IngestChecker checker;
+    private final CryptoService cryptoService;
+    private final ThrottleAccumulator throttleAccumulator;
+    private final SubmissionManager submissionManager;
+    private final HapiOpCounters opCounters;
+
+    /**
+     * Constructor of {@code IngestWorkflowImpl}
+     *
+     * @param nodeInfo the {@link NodeInfo} of the current node
+     * @param currentPlatformStatus the {@link CurrentPlatformStatus}
+     * @param stateAccessor a {@link Supplier} that provides the latest immutable state
+     * @param onset the {@link WorkflowOnset} that pre-processes the {@link ByteBuffer} of a
+     *     transaction
+     * @param checker the {@link IngestWorkflow} with specific checks of an ingest-workflow
+     * @param throttleAccumulator the {@link ThrottleAccumulator} for throttling
+     * @param submissionManager the {@link SubmissionManager} to submit transactions to the platform
+     * @param opCounters the {@link HapiOpCounters} with workflow-specific metrics
+     */
+    public IngestWorkflowImpl(
+            @NonNull final NodeInfo nodeInfo,
+            @NonNull final CurrentPlatformStatus currentPlatformStatus,
+            @NonNull final Supplier<AutoCloseableWrapper<HederaState>> stateAccessor,
+            @NonNull final WorkflowOnset onset,
+            @NonNull final IngestChecker checker,
+            @NonNull final CryptoService cryptoService,
+            @NonNull final ThrottleAccumulator throttleAccumulator,
+            @NonNull final SubmissionManager submissionManager,
+            @NonNull final HapiOpCounters opCounters) {
+        this.nodeInfo = requireNonNull(nodeInfo);
+        this.currentPlatformStatus = requireNonNull(currentPlatformStatus);
+        this.stateAccessor = requireNonNull(stateAccessor);
+        this.onset = requireNonNull(onset);
+        this.checker = requireNonNull(checker);
+        this.cryptoService = requireNonNull(cryptoService);
+        this.throttleAccumulator = requireNonNull(throttleAccumulator);
+        this.submissionManager = requireNonNull(submissionManager);
+        this.opCounters = requireNonNull(opCounters);
+    }
+
     @Override
-    public void handleTransaction(
-            @NonNull final SessionContext session,
+    public void submitTransaction(
+            @NonNull final SessionContext ctx,
             @NonNull final ByteBuffer requestBuffer,
             @NonNull final ByteBuffer responseBuffer) {
-        // Implementation to be completed by Issue #4209
+
+        ResponseCodeEnum result = OK;
+        long estimatedFee = 0L;
+
+        // Do some general pre-checks
+        if (nodeInfo.isSelfZeroStake()) {
+            result = INVALID_NODE_ACCOUNT;
+        }
+        if (currentPlatformStatus.get() != ACTIVE) {
+            result = PLATFORM_NOT_ACTIVE;
+        }
```

Comment:
Is it true that we want the result PLATFORM_NOT_ACTIVE to take precedence over INVALID_NODE_ACCOUNT? Maybe if/else instead of two if checks to prevent that kind of ambiguity?